### ISSUE TITLE
Supporting Container Images in Mesos Containerizer doesn't work by using marathon api

### ISSUE THREAD

**Issue by wangqunOne** [mesosphere/marathon#3716]
Hi
I use the marathon api to create tasks to test Supporting Container Images in Mesos Containerizer .
My steps is the following:
1) to run the process in master node.
`sudo /usr/sbin/mesos-master --zk=zk://10.0.0.4:2181/mesos --port=5050 --log_dir=/var/log/mesos --cluster=mesosbay --hostname=10.0.0.4 --ip=10.0.0.4 --quorum=1 --work_dir=/var/lib/mesos`
2) to run the process in slave node.
`sudo /usr/sbin/mesos-slave --master=zk://10.0.0.4:2181/mesos --log_dir=/var/log/mesos --containerizers=docker,mesos --executor_registration_timeout=5mins --hostname=10.0.0.5 --ip=10.0.0.5 --isolation=docker/runtime,filesystem/linux --work_dir=/tmp/mesos/slave --image_providers=docker --executor_environment_variables="{}"`
3) to create one json file to specify the container to be managed by mesos.
`sudo touch mesos.json`
`sudo vim mesos.json`

```
{
  "container": {
    "type": "MESOS",
    "mesos": {
        "image": {
            "type": "DOCKER",
            "docker": {
                "name": "library/redis"
            }
        }
    }
  },
  "id": "ubuntumesos",
  "instances": 1,
  "cpus": 0.5,
  "mem": 512,
  "uris": [],
  "cmd": "ping 8.8.8.8"
}
```

4) `sudo curl -X POST -H "Content-Type: application/json" localhost:8080/v2/apps -d@mesos.json`
5) `sudo curl http://localhost:8080/v2/tasks
`{"tasks":[
{"id":"ubuntumesos.fc1879be-fc9f-11e5-81e0-024294de4967","host":"10.0.0.5","ipAddresses":[],"ports":[31597],"startedAt":"2016-04-07T09:06:24.900Z","stagedAt":"2016-04-07T09:06:16.611Z","version":"2016-04-07T09:06:14.354Z","slaveId":"058fb5a7-9273-4bfa-83bb-8cb091621e19-S1","appId":"/ubuntumesos","servicePorts":[10000]}]}`
6) I open the slave workdir "/tmp/mesos/store/docker" that stores mesos image and don't find the image to be pulled.

**Comment by robbrockbank** [comment]
You aren't using the Docker containerizer, so you won't be able to access the container using the Docker client.

**Comment by wangqunOne** [comment]
@robbrockbank I understand your means that Marathon doesn't support the container using the Unified Containerizer, Does Marathon community  have plan to support the feature? Thank you.

**Comment by wangqunOne** [comment]
@philwinder Can you tell me how to know container of type MESOS status if it has been created successfully. Any command ...? Because I aware that I am using the mesos containerizer, if I shouldn't access the docker client by command "sudo docker run -ti --net=host redis redis-cli". However, I don't know how to validate it?

**Comment by wangqunOne** [comment]
@philipnrmn

**Comment by philipnrmn** [comment]
@wangqunOne as @robbrockbank says, your mesos.json file is incorrect. If you want to start a docker container, you must specify `"type": "DOCKER"` not `"type": "MESOS"`. See the [simple docker recipe](http://mesosphere.github.io/marathon/docs/recipes.html) in the Marathon docs for an example. 

Additionally, you don't need to specify a ping command. Try the following definition instead:

```
{
  "container": {
    "type": "DOCKER",
    "docker": {
      "image": "redis"
    }
  },
  "id": "mesos-docker-redis",
  "instances": 1,
  "cpus": 0.5,
  "mem": 512
}
```

Finally, please note that you should check the response from the app creation endpoint to see whether your app configuration was accepted. Sending the configuration you posted in your OP results in the following error message:
`
{
  "message": "Object is not valid",
  "details": [
    {
      "path": "/container/docker",
      "errors": [
        "must be empty"
      ]
    }
  ]
}
`

This might have helped you diagnose your own issue.

**Comment by wangqunOne** [comment]
@philipnrmn You are wrong. I aren't using the Docker containerizer but to be Mesos containerizer, So I should specify "type": "Mesos" not "type": "Docker". I find the marathon doesn't support the mesos image to pull the work_dir.
 {
  "container": {
    "type": "MESOS",
    "mesos": {
        "image": {
            "type": "DOCKER",
            "docker": {
                "name": "library/redis"
            }
        }
    }
  },
  "id": "ubuntumesos",
  "instances": 1,
  "cpus": 0.5,
  "mem": 512,
  "uris": [],
  "cmd": "ping 8.8.8.8"
}

**Comment by HackToday** [comment]
hi @robbrockbank  and @philipnrmn  in Short, 
https://github.com/apache/mesos/blob/master/docs/container-image.md#supporting-container-images-in-mesos-containerizer

Mesos have `Docker containerizer` and `Mesos containerizer` (the above link said it was improved named Unified Containerizer)

So does Marathon community support Mesos containerizer ?  If not, what's the plan for that case ?

**Comment by wangqunOne** [comment]
@aquamatthias

**Comment by wangqunOne** [comment]
@mlunoe

**Comment by philipnrmn** [comment]
@HackToday thanks for the clarification, I understand the question now. No, Marathon does not presently support those features of the unified containerizer (which is still marked as experimental by Mesos). Pull requests are welcome! 

@wangqunOne I don't believe I am wrong. The application configuration you specified will not be accepted by the API. If you want to use the redis docker image, you should use the docker containerizer.

**Comment by tmjd** [comment]
With the recent release of Mesos 1.0.0-rc1 is there any work being done on supporting the Mesos containerizer for docker images?  I don't believe it will be considered experimental that that point.  If no work is being done yet, are there plans to begin work on this feature?

**Comment by aquamatthias** [comment]
@tmjd we are currently working on supporting the Mesos universal containerizer.

**Comment by cheuschober** [comment]
@aquamatthias For those of us who aren't in a position to wait for an official implementation and are likely to fork to support it, is there a branch that is acting as a reference for this feature?

**Comment by Sun-zhe** [comment]
@aquamatthias How is the Mesos universal containerizer going on? Can I use Mesos container and Docker images to execute my own instance?

**Comment by xiaods** [comment]
Mesos 1.0 is out, any update on marathon side?

**Comment by aquamatthias** [comment]
PR is here: https://github.com/mesosphere/marathon/pull/4088 currently in review/test.

**Comment by xiaods** [comment]
this issue should be closed by #4088

**Comment by meichstedt** [comment]
Note: This issue has been migrated to https://jira.mesosphere.com/browse/MARATHON-4660. For more information see https://groups.google.com/forum/#!topic/marathon-framework/khtvf-ifnp8.
